### PR TITLE
Add json ouptut to reqtraq validate

### PR DESCRIPTION
**Pull request by Javier-varez**
To be used with Phabricator lint to receive inline messages in the
review.

### CODE REVIEW

File: validate_cmd.go [hunk 1]
```diff
@@ -18,26 +23,59 @@ var validateCmd = &cobra.Command{
...
+func buildJsonIssues(issues []Issue, jsonWriter *json.Encoder) {
+	for _, issue := range issues {
+		// Only report issues for the current repository
+		if issue.RepoName != repos.BaseRepoName() {
+			continue
+		}
+
+		jsonWriter.Encode(LintMessage{
+			Name:        "Reqtraq",
+			Code:        "Reqtraq",
```
Comment (finomen) [inline validate_cmd.go:48]: Why all errors with same name? It'll be ugly in phabricator

**Review by finomen — CHANGES_REQUESTED** [review]

File: validate_cmd.go (hunk 1 above)
Comment (Javier-varez) [reply validate_cmd.go:48]: As far as I understand we have the same for clang-format:
"name":"code must be clang-formatted",

I'm open to suggestions about what to use instead, but the error will be in the message

**Review by finomen — APPROVED** [review]

**Comment by Javier-varez** [comment]
Rebased on master

**Review by dawzee — CHANGES_REQUESTED** [review]

File: req.go [hunk 2]
```diff
@@ -168,23 +189,44 @@ func (rg *ReqGraph) resolve() []error {
 
 		// Validate requirement Id
 		if !req.Document.Schema.Requirements.MatchString(req.ID) {
-			errs = append(errs, fmt.Errorf("Requirement `%s` in document `%s` does not match required regexp `%s`", req.ID, req.Document.Path, req.Document.Schema.Requirements))
+			issue := Issue{
+				Line:     req.Position,
```
[the remark this answers is not in the document]
Comment (Javier-varez) [reply req.go:193]: Oh, very good point, I did not realize it worked like that

**Review by dawzee — APPROVED** [review]

**Pull request merged**